service/collector.go

```diff
@@ -111,12 +113,16 @@ func (col *Collector) GetLogger() *zap.Logger {
 
 // Shutdown shuts down the collector server.
 func (col *Collector) Shutdown() {
-	defer func() {
-		if r := recover(); r != nil {
-			col.logger.Info("shutdownChan already closed")
-		}
-	}()
-	close(col.shutdownChan)
+	// Only shutdown if we're in a Running or Starting State else noop
+	state := col.GetState()
+	if state == Running || state == Starting {
```

Comment:
i think the state here could also be `Closing` no? 

https://github.com/open-telemetry/opentelemetry-collector/blob/26e62ce0642e728943b7aa6cdbcd687761d3932a/service/collector.go#L144-L148

Reply:
Maybe, that specific set of code is for restarting. I wonder if we should have another `Restarting` state. I omitted `Closing` because the state get's set to `Closing` also in the `shutdown` func that is trigger by closing the `shutdownChan`. Which means another call to `Shutdown` in that still will trigger the panic/recover.

https://github.com/open-telemetry/opentelemetry-collector/blob/26e62ce0642e728943b7aa6cdbcd687761d3932a/service/collector.go#L238-L239

I do like the idea of some state that indicates the collector is restarting/reconfiguring. As a user seeing those states coming from a black box it makes more sense then just seeing `Closing` -> `Starting` -> `Running`.

Thoughts?

Reply:
@cpheps please see this comment. Still not sure why not the same behavior as the `http.Server`

Reply:
Isn't this the same behavior? If you call shutdown before the collector has started, `Starting` state, it'll shut down as soon as `Run` enters the `select` statement. If it's in the `Running` state it'll shutdown. Any other state will just cause the channel close to panic and the recover statement to process and that doesn't seem worth it if we know it can't succeed.

Calling shutdown in `http.Server` before its listening will cause any listen call to close immediately. Let me know if I'm misunderstanding something.

Reply:
Probably in a separate PR, but we should block for shutdown to return when complete correct? Also RUN should probably not start everything if shutdown was called.

We can improve this, probably separate issue to track all of these.